Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM python:3.7-slim-stretch
+
+# Many of these dependencies are required to build the app's dependencies, so staging these out doesn't help much
+RUN mkdir -p /app
+RUN apt-get update && apt-get install -y git gcc clang cmake pkg-config libdbus-1-dev libglib2.0-dev libcairo2-dev python3-dev libgirepository1.0-dev wget
+
+WORKDIR /app
+RUN wget https://gitlab.matrix.org/matrix-org/olm/-/archive/master/olm-master.tar.bz2 \
+    && tar -xvf olm-master.tar.bz2 \
+    && cd olm-master && make && make PREFIX="/usr" install && cd ../ \
+    && rm -r olm-master
+
+COPY . /app
+RUN pip install . gobject PyGObject "matrix-nio@git+https://github.com/poljar/matrix-nio.git@dev#egg=matrix-nio-0" && python setup.py install
```

Comment:
The description of gobject is "Pythonic Google Geocode API wrapper", I don't think we need this.

 PyGObject is nowadays optional and only needed for the UI, there is a bug around this being optional but it may be Mac specific.

matrix-nio nowadays lives on pypi, no need to fetch it from github.

Reply:
gobject used to be required to build the thing, but empirically not anymore. PyGObject is still needed (ImportError: no module named 'gi' without)